presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -199,9 +182,32 @@ public int getPositionCount()
     @Override
     public int getSizeInBytes()
     {
+        if (sizeInBytes < 0) {
```

Comment:
In theory, this means another block may never see it. How is it like in real life? Does this deliberate racy access do more benefit than harm?